pkg/kubeadm/cmd/cmd.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"io"
+
+	"github.com/renstrom/dedent"
+	"github.com/spf13/cobra"
+
+	kubeadmapi "k8s.io/kubernetes/pkg/kubeadm/api"
+)
+
+func NewKubeadmCommand(in io.Reader, out, err io.Writer, envParams map[string]string) *cobra.Command {
+	cmds := &cobra.Command{
+		Use:   "kubeadm",
+		Short: "kubeadm: bootstrap a secure kubernetes cluster easily.",
+		Long: dedent.Dedent(`
+			kubeadm: bootstrap a secure kubernetes cluster easily.
+
+			    ┌──────────────────────────────────────────────────────────┐
+			    │ KUBEADM IS ALPHA, DO NOT USE IT FOR PRODUCTION CLUSTERS! │
+			    │                                                          │
+			    │ But, please try it out! Give us feedback at:             │
+			    │ https://github.com/kubernetes/kubernetes/issues          │
+			    │ and at-mention @kubernetes/sig-cluster-lifecycle         │
+			    └──────────────────────────────────────────────────────────┘
+
+			Example usage:
+
+			    Create a two-machine cluster with one master (which controls the cluster),
+			    and one node (where workloads, like pods and replica sets run).
+
+			    ┌──────────────────────────────────────────────────────────┐
+			    │  On the first machine                                    │
+			    ├──────────────────────────────────────────────────────────┤
+			    │ master# kubeadm init master                              │
+			    │ Your token is: <token>                                   │
+			    └──────────────────────────────────────────────────────────┘
+
+			    ┌──────────────────────────────────────────────────────────┐
+			    │ On the second machine                                    │
+			    ├──────────────────────────────────────────────────────────┤
+			    │ node# kubeadm join node --token=<token> <ip-of-master>   │
+			    └──────────────────────────────────────────────────────────┘
+
+			    You can then repeat the second step on as many other machines as you like.
+
+		`),
+	}
+	// TODO find a way to set master node `Unschedulable`, i.e.
```

Comment:
Yeah, bit here we can ensure what kubelet flags will be, right? This to-do
note is slightly out of date. I have read your PR yesterday, but wasn't
sure what exactly I need to do via the API, is there a helper or I need to
fetch node object, update it and apply the changes?

On Wed, 31 Aug 2016, 21:58 Mike Danese, <notifications@github.com> wrote:

> In pkg/kubeadm/cmd/cmd.go
> <https://github.com/kubernetes/kubernetes/pull/31221#discussion_r77072137>
> :
>
> > +			    ├──────────────────────────────────────────────────────────┤
> > +			    │ master# kubeadm init master                              │
> > +			    │ Your token is: <token>                                   │
> > +			    └──────────────────────────────────────────────────────────┘
> > +
> > +			    ┌──────────────────────────────────────────────────────────┐
> > +			    │ On the second machine                                    │
> > +			    ├──────────────────────────────────────────────────────────┤
> > +			    │ node# kubeadm join node --token=<token> <ip-of-master>   │
> > +			    └──────────────────────────────────────────────────────────┘
> > +
> > +			    You can then repeat the second step on as many other machines as you like.
> > +
> > +		`),
> > +	}
> > +	// TODO find a way to set master node `Unschedulable`, i.e.
>
> Yes but it's deployment specific. #31647
> <https://github.com/kubernetes/kubernetes/pull/31647>
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/31221/files/57096ca902c5c666d45df191037d9278972af1fd#r77072137>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAPWSwbnNM8sYfLcbFQktRkrW9138SjIks5qlesOgaJpZM4JqsTH>
> .
>